firmware/defmt-rtt/src/lib.rs

```diff
@@ -206,3 +223,26 @@ unsafe fn handle() -> &'static Channel {
 
     &_SEGGER_RTT.up_channel
 }
+
+fn host_is_connected(channel: &Channel) -> bool {
+    channel.flags.load(Ordering::Relaxed) == BLOCK_IF_FULL
+}
+
+/// Busy wait, until the read- catches up with the write-pointer.
+///
+/// Returns early, if the write-pointer stops progressing.
+fn busy_wait(read: impl Fn() -> usize, write: impl Fn() -> usize) {
+    let mut prev_read = 0;
+
+    while read() != write() {
+        // wait for some CPU cycles
+        asm::delay(100); // TODO: what's a useful number?
+
+        // did the write-pointer progress?
+        // if it didn't the host is either disconnected or too slow
+        match read() == prev_read {
+            true => return,
+            false => prev_read = read(),
+        }
```

Comment:
Do we want to catch if the host disconnects while flushing? Else we could drop the whole while-body.